frontend-react/src/utils/misc.ts

```diff
@@ -13,3 +15,43 @@ export const splitOn: {
     <T extends any[]>(s: T, ...i: number[]): T[];
 } = <T>(slicable: string | T[], ...indices: number[]) =>
     [0, ...indices].map((n, i, m) => slicable.slice(n, m[i + 1]));
+
+/**
+ *
+ * @param jsonTextValue string to be json.parsed
+ * @param elemLabel used when displaying error in case there are multiple textarea's on the page
+ * @param textInputRef used to select the range of text where the error happened.
+ * @return false if fails to parse or the object of the successfully parsed json
+ */
+export const checkTextAreaJson = (
+    jsonTextValue: string,
+    elemLabel: string,
+    textInputRef: React.RefObject<HTMLTextAreaElement>
+): false | Object => {
+    try {
+        return JSON.parse(jsonTextValue);
+    } catch (err: any) {
+        // message like `'Unexpected token _ in JSON at position 164'`
+        // or           `Unexpected end of JSON input`
+        const errMsg = err?.message || "unknown error";
+        showError(`Element "${elemLabel}" generated an error "${errMsg}"`);
+
+        // now we parse out the position and try to select it for them.
+        // NOTE: if "at position N" string not found, then assume mistake is at the end
+        let errStartOffset = jsonTextValue.length;
+        const findPositionMatch = errMsg?.matchAll(/position (\d+)/gi)?.next();
+        if (findPositionMatch?.value?.length === 2) {
+            const offset = parseInt(findPositionMatch.value[1] || -1);
+            if (!isNaN(offset) && offset !== -1) {
+                errStartOffset = offset;
+            }
+        }
+
+        // now select the problem area inside the TextArea
+        const errEndOffset = Math.min(errStartOffset + 4, jsonTextValue.length); // don't let go past len
```

Comment:
Curious, why is the end offset the start + 4? Also why `findPositionMatch` compared with 2? I was trying to understand how the original function was working

Reply:
It just selects around the mistake. The json parser can be fooled pretty easily into thinking the mistake is somewhere it
it's not, so I'm just selecting +/- 4 characters so it's a little less misleading.